improver_tests/calibration/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py

```diff
@@ -475,36 +408,25 @@ def test_end_to_end(self):
     @ManageWarnings(
         ignored_messages=["Collapsing a non-contiguous coordinate."])
     def test_end_to_end_with_mask(self):
-        """An example end-to-end calculation, this time using a mask to limit
-        the application of calibration to regions where the mask=1."""
+        """An example end-to-end calculation, but making sure that the
+        areas that are masked within the landsea mask, are masked at the
+        end."""
 
         # Construct a mask and encapsulate as a cube.
-        mask = np.ones((3, 3))
-        mask[1:, 1:] = 0
+        mask = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
         mask_cube = self.current_temperature_forecast_cube[0].copy(data=mask)
-
-        # Collapse input realizations to get uncalibrated expectation values.
-        forecast_predictors = self.current_temperature_forecast_cube.collapsed(
-            "realization", iris.analysis.MEAN)
-        forecast_vars = self.current_temperature_forecast_cube.collapsed(
-            "realization", iris.analysis.VARIANCE)
-
-        # Manually construct merged calibrated and uncalibrated arrays.
-        self.expected_loc_param_mean[1:, 1:] = (
-            forecast_predictors.data[1:, 1:])
-        self.expected_scale_param_mean[1:, 1:] = (
-            forecast_vars.data[1:, 1:])
+        mask_loc_param = np.ma.masked_where(mask, self.expected_loc_param_mean)
+        mask_scale_param = np.ma.masked_where(
+            mask, self.expected_scale_param_mean)
 
         calibrated_forecast_predictor, calibrated_forecast_var = (
             self.plugin.process(self.current_temperature_forecast_cube,
                                 self.coeffs_from_mean, landsea_mask=mask_cube))
 
         self.assertCalibratedVariablesAlmostEqual(
-            calibrated_forecast_predictor.data,
-            self.expected_loc_param_mean)
+            calibrated_forecast_predictor.data, mask_loc_param)
```

Comment:
Does this check that the mask and the data are both the same?

Reply:
I am not sure this is right. I think I have flipped the mask and this still passes, so this needs to be rewritten I think.

Reply:
I've change this so it checks the array.data and array.mask separately.